app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -249,6 +249,8 @@ protected void deliverUpdate(LoginActivity receiver, Integer... update) {
                             receiver.updateProgress(Localization.get("sync.progress.downloading"), DataPullTask.DATA_PULL_TASK_ID);
                         } else if (update[0] == DataPullTask.PROGRESS_DOWNLOADING) {
                             receiver.updateProgress(Localization.get("sync.process.downloading.progress", new String[]{String.valueOf(update[1])}), DataPullTask.DATA_PULL_TASK_ID);
+                        } else if (update[0] == DataPullTask.PROGRESS_DOWNLOADING_COMPLETE) {
+                            receiver.hideTaskCancelButton();

```

Comment:
Before this PR the login data pull task could be cancelled, but it wouldn't do anything: just wait for the normal code to finish. Now that we can abort http requests, cancelling is actually useful because it aborts the login http request faster. Also provides same guarantee as syncing that task won't be cancelled while data is being commited.